commands/volumes/volume-status.go

```diff
@@ -0,0 +1,142 @@
+package volumecommands
+
+import (
+	goerrors "errors"
+	"net/http"
+
+	"github.com/gluster/glusterd2/brick"
+	"github.com/gluster/glusterd2/errors"
+	"github.com/gluster/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/rest"
+	"github.com/gluster/glusterd2/transaction"
+	"github.com/gluster/glusterd2/volume"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func checkStatus(ctx transaction.TxnCtx) error {
+	var volname string
+
+	if err := ctx.Get("volname", &volname); err != nil {
+		ctx.Logger().WithFields(log.Fields{
+			"error": err,
+			"key":   "volname",
+		}).Error("checkStatus: Failed to get key from transaction context.")
+		return err
+	}
+
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		ctx.Logger().WithFields(log.Fields{
+			"error": err,
+			"key":   "volname",
+		}).Error("checkStatus: Failed to get volume information from store.")
+		return err
+	}
+
+	var brickStatuses []*brick.Brickstatus
+
+	for _, binfo := range vol.Bricks {
+		// Skip bricks that aren't on this node.
+		// TODO: Rename Brickinfo field 'ID' to 'NodeUUID'
+		if uuid.Equal(binfo.ID, gdctx.MyUUID) == false {
+			continue
+		}
+
+		// TODO: Check actual brick status when we get them running.
+		fakeStatus := &brick.Brickstatus{
+			Hostname: binfo.Hostname,
+			Path:     binfo.Path,
+			ID:       binfo.ID,
+			Online:   false,
+			Pid:      1234,
+		}
+		brickStatuses = append(brickStatuses, fakeStatus)
+	}
+
+	// Store the results in transaction context. This will consumed by the
+	// node that initiated the transaction.
+	key := gdctx.MyUUID.String() + "/" + "brickstatuses"
```

Comment:
It'll be better if we can provide a properly defined method for this. We wouldn't want users to be creating arbitrary keys here. Some thing like `TxnCtx.SetMyResult(key, value)` which would set the value with the key "MyUUID/<key>". And a complementary `TxnCtx.GetResult(uuid, key)`.